From edb95a6a610e4780efb98365296dc5d2740e0f9c Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sat, 17 Jan 2015 19:08:57 -0800
Subject: [PATCH 1007/1009] Only call setpriority(PRIO_DARWIN_THREAD, 0,
 PRIO_DARWIN_BG) if it is available

Tiger and earlier versions of darwin do not support this.

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit 92a47c4641be0c3fbc0e660f5680b3d4d524887f)
---
 tools/libclang/CIndex.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git llvm_release_36/tools/clang/tools/libclang/CIndex.cpp macports_release_36/tools/clang/tools/libclang/CIndex.cpp
index 00ef8c0..6f124be 100644
--- llvm_release_36/tools/clang/tools/libclang/CIndex.cpp
+++ macports_release_36/tools/clang/tools/libclang/CIndex.cpp
@@ -7044,7 +7044,7 @@ void clang::setThreadBackgroundPriority() {
   if (getenv("LIBCLANG_BGPRIO_DISABLE"))
     return;
 
-#ifdef USE_DARWIN_THREADS
+#if defined(USE_DARWIN_THREADS) && defined(PRIO_DARWIN_THREAD) && defined(PRIO_DARWIN_BG)
   setpriority(PRIO_DARWIN_THREAD, 0, PRIO_DARWIN_BG);
 #endif
 }
-- 
2.10.1

